### DISCUSSION TITLE
Is there a way to detect the main thread during Uvicorn reloads?

### DISCUSSION THREAD

**Discussion by browser-bug** [Kludex/uvicorn#1875] [Q&A] [answered]
Hello, I'm looking for a way to detect the main thread during Uvicorn reloads. As well explained here https://stackoverflow.com/questions/62766539/uvicorn-detect-reload-main-thread , is there something like `RUN_MAIN` or `WERKZEUG_RUN_MAIN` environment variables? Since i'm attaching a debugpy listener I'd like to avoid restarting it everytime some file changes are detected. Thanks in advance.

**Comment by Kludex** [comment, marked as answer]
There are two processes: the _reloader_ and the server. The _reloading process_ is not via threads.

**Reply by browser-bug** [reply]
Meaning that there's no way to avoid the debugpy listener to respawn everytime?

**Reply by Kludex** [reply]
There's no way using the native _reloader_ feature from `uvicorn`.
